diff -Naur spice/server/char-device.c new_spice/server/char-device.c
--- spice/server/char-device.c	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/char-device.c	2017-02-19 02:11:13.309802076 +0100
@@ -22,8 +22,8 @@
 
 #include <config.h>
 #include <inttypes.h>
-#include "char-device.h"
 #include "red-client.h"
+#include "char-device.h"
 #include "reds.h"
 #include "glib-compat.h"
 
diff -Naur spice/server/dcc.h new_spice/server/dcc.h
--- spice/server/dcc.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/dcc.h	2017-02-19 09:32:44.225279860 +0100
@@ -80,8 +80,13 @@
 
 typedef struct DisplayChannel DisplayChannel;
 typedef struct Stream Stream;
+
+
+
 typedef struct StreamAgent StreamAgent;
 
+
+
 typedef struct WaitForChannels {
     SpiceMsgWaitForChannels header;
     SpiceWaitForChannel buf[MAX_CACHE_CLIENTS];
diff -Naur spice/server/dcc-private.h new_spice/server/dcc-private.h
--- spice/server/dcc-private.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/dcc-private.h	2017-02-19 09:36:03.882625425 +0100
@@ -24,7 +24,6 @@
 #include "stream.h"
 #include "red-channel-client.h"
 
-typedef struct DisplayChannelClientPrivate DisplayChannelClientPrivate;
 struct DisplayChannelClientPrivate
 {
     uint32_t id;
diff -Naur spice/server/display-channel.h new_spice/server/display-channel.h
--- spice/server/display-channel.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/display-channel.h	2017-02-19 02:35:05.716235523 +0100
@@ -57,7 +57,6 @@
 #define DISPLAY_CHANNEL_GET_CLASS(obj) \
     (G_TYPE_INSTANCE_GET_CLASS((obj), TYPE_DISPLAY_CHANNEL, DisplayChannelClass))
 
-typedef struct DisplayChannel DisplayChannel;
 typedef struct DisplayChannelClass DisplayChannelClass;
 typedef struct DisplayChannelPrivate DisplayChannelPrivate;
 
diff -Naur spice/server/main-channel-client.h new_spice/server/main-channel-client.h
--- spice/server/main-channel-client.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/main-channel-client.h	2017-02-19 02:25:39.324655417 +0100
@@ -38,7 +38,6 @@
 #define MAIN_CHANNEL_CLIENT_GET_CLASS(obj) \
     (G_TYPE_INSTANCE_GET_CLASS((obj), TYPE_MAIN_CHANNEL_CLIENT, MainChannelClientClass))
 
-typedef struct MainChannelClient MainChannelClient;
 typedef struct MainChannelClientClass MainChannelClientClass;
 typedef struct MainChannelClientPrivate MainChannelClientPrivate;
 
diff -Naur spice/server/red-channel-client.h new_spice/server/red-channel-client.h
--- spice/server/red-channel-client.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/red-channel-client.h	2017-02-19 02:21:19.934872809 +0100
@@ -41,7 +41,6 @@
 #define RED_CHANNEL_CLIENT_GET_CLASS(obj) \
     (G_TYPE_INSTANCE_GET_CLASS((obj), RED_TYPE_CHANNEL_CLIENT, RedChannelClientClass))
 
-typedef struct RedChannelClient RedChannelClient;
 typedef struct RedChannelClientClass RedChannelClientClass;
 typedef struct RedChannelClientPrivate RedChannelClientPrivate;
 
diff -Naur spice/server/red-channel.h new_spice/server/red-channel.h
--- spice/server/red-channel.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/red-channel.h	2017-02-19 02:21:39.543012623 +0100
@@ -37,9 +37,9 @@
 
 G_BEGIN_DECLS
 
+typedef struct RedClient RedClient;
 typedef struct RedChannel RedChannel;
 typedef struct RedChannelClient RedChannelClient;
-typedef struct RedClient RedClient;
 typedef struct MainChannelClient MainChannelClient;
 
 typedef uint8_t *(*channel_alloc_msg_recv_buf_proc)(RedChannelClient *channel,
diff -Naur spice/server/red-client.h new_spice/server/red-client.h
--- spice/server/red-client.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/red-client.h	2017-02-19 02:18:20.629543323 +0100
@@ -34,7 +34,6 @@
 #define RED_CLIENT_GET_CLASS(obj) \
     (G_TYPE_INSTANCE_GET_CLASS ((obj), RED_TYPE_CLIENT, RedClientClass))
 
-typedef struct RedClient RedClient;
 typedef struct RedClientClass RedClientClass;
 
 GType red_client_get_type (void) G_GNUC_CONST;
diff -Naur spice/server/stream.h new_spice/server/stream.h
--- spice/server/stream.h	2017-02-19 02:00:39.000000000 +0100
+++ new_spice/server/stream.h	2017-02-19 09:28:10.719575373 +0100
@@ -42,7 +42,6 @@
 #define RED_STREAM_DEFAULT_LOW_START_BIT_RATE (2.5 * 1024 * 1024) // 2.5Mbps
 #define MAX_FPS 30
 
-typedef struct Stream Stream;
 
 typedef struct RedStreamActivateReportItem {
     RedPipeItem pipe_item;
@@ -62,7 +61,7 @@
 } StreamStats;
 #endif
 
-typedef struct StreamAgent {
+struct StreamAgent {
     QRegion vis_region; /* the part of the surface area that is currently occupied by video
                            fragments */
     QRegion clip;       /* the current video clipping. It can be different from vis_region:
@@ -83,7 +82,7 @@
 #ifdef STREAM_STATS
     StreamStats stats;
 #endif
-} StreamAgent;
+};
 
 typedef struct RedStreamClipItem {
     RedPipeItem base;
