diff -Naur spice-gtk/tools/Makefile.am new_spice-gtk/tools/Makefile.am
--- spice-gtk/tools/Makefile.am	2017-02-20 23:31:36.565920194 +0100
+++ new_spice-gtk/tools/Makefile.am	2017-02-19 00:12:19.000000000 +0100
@@ -28,6 +28,7 @@
 spicy_LDADD =				\
 	$(top_builddir)/src/libspice-client-gtk-3.0.la	\
 	$(top_builddir)/src/libspice-client-glib-2.0.la	\
+	$(GTK_LIBS) \
 	$(NULL)
 
 # FIXME: GtkAction and lots of GtkUIManager APIs are deprecated
@@ -60,6 +61,7 @@
 
 spicy_stats_LDADD =			\
 	$(top_builddir)/src/libspice-client-glib-2.0.la	\
+	$(GOBJECT2_LIBS)		\
 	$(NULL)
 
 spicy_stats_CPPFLAGS =			\
